controllers/managedocs_controller.go

```diff
@@ -678,8 +678,13 @@ func (r *ManagedOCSReconciler) reconcileAlertmanager() error {
 		}
 
 		desired := templates.AlertmanagerTemplate.DeepCopy()
-		r.alertmanager.ObjectMeta.Labels = map[string]string{monLabelKey: monLabelValue}
+		desired.Spec.AlertmanagerConfigSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				monLabelKey: monLabelValue,
+			},
+		}
 		r.alertmanager.Spec = desired.Spec
+		r.alertmanager.ObjectMeta.Labels = map[string]string{monLabelKey: monLabelValue}
```

Comment:
This will override all labels on the alertmanager which means that all other labels will be lost. 
Please add the label to an existing `ObjectMeta.labels` instead